src/hooks/useOutsideClickRef.ts

```diff
@@ -25,9 +25,10 @@ function useOutsideClickRef(
 
   const memoizedCallback = useCallback(
     (e: MouseEvent) => {
-      if (node && !node.contains(e.target as Element)) {
-        savedHandler.current(e);
+      if (!node || node.contains(e.target as Element)) {
```

Comment:
Does this fix the issue? 🤔

Reply:
No it does not, I realised it's the same logic as before

Reply:
Yeah we need to figure out what the cause of the bug is. Perhaps `currentTarget` might be a solution?

Reply:
Yes, the cause of the bug is that we did not use event capturing.

When user click a descendent element inside the ref, the default behaviour was that the `e.target` node's click event handler being called first, and if this changed the child element inside, the child will be updated immediately. After this, the click event propagates up, until document's click event handler being called, at that time, the `node` saved in our hook  is not up-to-date, the child element inside it is the old one, so in our `memoizedCallback`, `!node.contains(e.target as Element)` will be `true`, that's the cause of the bug.

So I changed to event capturing by adding `true` to `document.addEventListener("click", memoizedCallback, true);`, this will trigger the `document`'s event handler first, then does down to the child elements 

[example](https://codesandbox.io/s/useoutsideclickref-forked-45qbn?file=/src/useOutClick.ts)

Reply:
Makes total sense @qiweiii . I think we should update `useOutsideClick` hook too in the same way.

Reply:
Shall I create another PR, or just add it here?